src/dataServices/dynamoDbBundleServiceHelper.test.ts

```diff
@@ -484,75 +534,210 @@ describe('sortBatchRequests', () => {
+    sinon.stub(ddbHelperReturnReadResource, 'getMostRecentUserReadableResource').callsFake(function stubbedGet() {
+        return Promise.resolve(readResource);
+    });
+
+    // resource does not exist
+    const ddbHelperResourceNotFound = new DynamoDbHelper(new AWS.DynamoDB());
+    sinon.stub(ddbHelperResourceNotFound, 'getMostRecentUserReadableResource').callsFake(function stubbedGet() {
+        throw new ResourceNotFoundError('Patient', 'read');
+    });
+
+    test('CRUD operations updateCreateSupported=false', async () => {
+        // read esource exists
+        const actualResponseReturnReadResource = await DynamoDbBundleServiceHelper.sortBatchRequests(
+            operations,
+            ddbHelperReturnReadResource,
+        );
+
+        expect(actualResponseReturnReadResource.batchReadWriteResponses).toMatchObject(
+            expectedBatchReadWriteResponsesResourceFound,
+        );
+        expect(actualResponseReturnReadResource.deleteRequests).toMatchObject(expectedDeleteRequests);
+        expect(actualResponseReturnReadResource.writeRequests).toMatchObject([
+            ...expectedCreateRequests,
+            ...expectedUpdateRequests,
+        ]);
+
+        // no resource exists - all operations should return the same results
+        const actualResponseNoResource = await DynamoDbBundleServiceHelper.sortBatchRequests(
+            operations,
+            ddbHelperResourceNotFound,
+        );
+        // if no resource is found only create will successfully run
+        expect(actualResponseNoResource.batchReadWriteResponses).toMatchObject(
+            expectedBatchReadWriteResponsesNoResource,
+        );
+
+        expect(actualResponseNoResource.deleteRequests).toMatchObject([]);
+
+        expect(actualResponseNoResource.writeRequests).toMatchObject([...expectedCreateRequests]);
+    });
+
+    test('CRUD operations updateCreateSupported=true', async () => {
+        const expectedCreateRequestsUpsert = [
             {
-                operation: 'update',
-                resource: readResource,
-                resourceType: 'Patient',
-                id: 'read',
+                PutRequest: {
+                    Item: DynamoDBConverter.marshall(readResource),
+                },
+                originalRequestIndex: 2,
             },
             {
-                operation: 'create',
-                resource: writeResource,
-                fullUrl: '',
-                resourceType: 'Patient',
-                id: 'write',
+                PutRequest: {
+                    Item: DynamoDBConverter.marshall(writeResource),
+                },
+                originalRequestIndex: 3,
             },
         ];
-
-        const expectedBatchReadWriteResponses: BatchReadWriteResponse[] = [
+        const expectedBatchReadWriteResponsesUpsert: BatchReadWriteResponse[] = [
             {
                 id: readResource.resource.id,
-                vid: '1',
+                vid: '0',
```

Comment:
The expected vid for a created resource should be 1. Is there a reason why these tests have all been changed to throw 404 Not Found, instead of creating new tests for the upsert? I think we should avoid changes to existing tests as much as possible so we can ensure there aren't breaking changes.

Reply:
This is the expected response for operation: read, on a resource that doesn't exists
All tests with 404 Not Found are added tests - the original tests are still there. I added them to the existing tests for when the resource doesn't exist. In such cases: When  updateCreateSupported=false, there should be some 404 responses when the resource is not found (on read, delete, and update). When updateCreateSupported=true, there should be 404 responseswhen the resource is not found on read and delete.

Reply:
Gotcha I got confused by the git addition/removals 😅